homeassistant/components/nuki/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""Doorsensor Support for the Nuki Lock."""
+
+import logging
+
+from pynuki import STATE_DOORSENSOR_OPENED
+
+from homeassistant.components.binary_sensor import DEVICE_CLASS_DOOR, BinarySensorEntity
+
+from . import NukiEntity
+from .const import (
+    ATTR_LOCK_DOORSENSOR_STATE,
+    ATTR_LOCK_DOORSENSOR_STATE_NAME,
+    ATTR_NUKI_ID,
+    DATA_COORDINATOR,
+    DATA_LOCKS,
+    DOMAIN as NUKI_DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Nuki lock binary sensor."""
+    data = hass.data[NUKI_DOMAIN][entry.entry_id]
+    coordinator = data[DATA_COORDINATOR]
+
+    entities = []
+
+    for lock in data[DATA_LOCKS]:
+        if lock.is_door_sensor_activated:
+            entities.extend([NukiDoorsensorEntity(coordinator, lock)])
+
+    async_add_entities(entities)
+
+
+class NukiDoorsensorEntity(NukiEntity, BinarySensorEntity):
+    """Representation of a Nuki Lock Doorsensor."""
+
+    @property
+    def name(self):
+        """Return the name of the lock."""
+        return self._nuki_device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._nuki_device.nuki_id}_doorsensor"
+
+    @property
+    def extra_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_NUKI_ID: self._nuki_device.nuki_id,
+            ATTR_LOCK_DOORSENSOR_STATE: self._nuki_device.door_sensor_state,
```

Comment:
Do we need to have the `ATTR_LOCK_DOORSENSOR_STATE` attribute? We already use this to represent open closed state of the entity.